src/main/java/com/atlassian/bitbucket/jenkins/internal/extensions/buildstatus/BitbucketPostBuildStatus.java

```diff
@@ -0,0 +1,49 @@
+package com.atlassian.bitbucket.jenkins.internal.extensions.buildstatus;
+
+import com.atlassian.bitbucket.jenkins.internal.client.BitbucketClientFactoryProvider;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentialsAdaptor;
+import com.atlassian.bitbucket.jenkins.internal.model.BitbucketBuildStatus;
+import com.google.inject.Injector;
+import hudson.model.AbstractBuild;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitSCM;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.gitclient.CheckoutCommand;
+import org.jenkinsci.plugins.gitclient.GitClient;
+
+import java.util.Optional;
+
+public class BitbucketPostBuildStatus extends GitSCMExtension {
+
+    private final String serverId;
+
+    public BitbucketPostBuildStatus(String serverId) {
+        this.serverId = serverId;
+    }
+
+    @Override
+    public void decorateCheckoutCommand(GitSCM scm, Run<?, ?> build, GitClient git, TaskListener listener,
+                                        CheckoutCommand cmd) throws GitException {
+        Injector injector = Optional.ofNullable(Jenkins.get().getInjector()).orElseThrow(RuntimeException::new);
```

Comment:
There are some really ugly null checks in this code. To remove the warnings I wrapped them in Optionals, given that if any of them fail, an exception feels appropriate (probably means either the Jenkins instance or the `BitbucketServerConfiguration` are incorrectly configured). Still, I'd prefer the command simply not execute in these cases- the build status should not be the one handling these issues. Open to suggestions on how to do this and satisfy the `@CheckForNull` requirements.

Reply:
Definitely prefer this approach